### DISCUSSION TITLE
how to define a moving wall boundary condition in MRT

### DISCUSSION THREAD

**Discussion by ztdepztdep** [CFD-GO/TCLB#367] [Q&A]
I want to compute the lid-driven cavity problem, but could you please tell me how to define a  moving wall boundary condition.

**Comment by llaniewski** [comment]
Quick search for "MovingWall" provides a nice example in one of the multiphase models:

https://github.com/CFD-GO/TCLB/blob/9e1ceb17cf2867311d4c1e08cd4cc2bbeb8c3815/models/multiphase/d2q9_pf_velocity/Dynamics.c.Rt#L1365-L1378

In this case `g[...]` are standard LBM densities, and the `S` in the function name stands for "South", so bottom wall. From this you should be able to construct the moving wall condition you need.

**Reply by ztdepztdep** [reply]
hello sir: 
   I have tried, it always gives inf value of density and velocity. I can't figure it out where is the problem.  I am wondering why this simple boundary condiction not be included in the standard d2q9 solver.

**Reply by ztdepztdep** [reply]
`CudaDeviceFunction void MovingWall_N_Dirichlet(){

	f[4] = f[2];
	f[7] = f[5] - WallX/6.0;
	f[8] = f[6] + WallX/6.0;
}

I added this piece of code into the code. But it cann't work. hope you can help me out of this.
`

**Reply by llaniewski** [reply]
@ztdepztdep I recommend doing the tutorial on model development (https://docs.tclb.io/tutorials/model-development/1.-finite-difference-wave-equation/) to understand how the note-types work. You'll learn how to add your new node types to `Run`, etc.

**Comment by ztdepztdep** [comment]
` 

       <?xml version="1.0"?>
       <CLBConfig version="2.0" output="output/" permissive="true">
	
    <Geometry nx="120" ny="120">
		<MRT>
			<Box/>
		</MRT>
		<MovingWall_N mask="ALL">
			<Box dy="-1"/>
		</MovingWall>

		<Wall mask="ALL">
	  		<Box nx="1"/>
			<Box dx="-1"/>
			<Box ny="1"/>
        </Wall>
        
	</Geometry>
    
	<Model>
		<Param name="VelocityX" value="1.0"/>
 		<Param name="Viscosity" value="0.12"/>
	</Model>
	<VTK Iterations="5000" what="U,Rho"/>
    <Log Iterations="5000" name="log.txt"/>
	<Solve Iterations="100000"/>
</CLBConfig>
`
  I added movingwall NodeType, and compiled it sucessfully. but when i run the xml file, it feed back the error about :
          error   ! Error while parsing drivencavity-2D.xml: Start-end tags mismatch

**Reply by TravisMitchell** [reply]
This is likely because you have "Start-end tags mismatch" for the MovingWall_N element in your xml.

**Reply by ztdepztdep** [reply]
```
<MovingWall_N mask="ALL">	
<Box dy="-1"/>	
</MovingWall>

```
 the north element is the last row of box ,as i have defined in the xml file. 
  i cann't figure it out what is the problem

**Reply by TravisMitchell** [reply]
You have <MovingWall_N> closed with </ MovingWall> ... These tags do not match.

**Reply by ztdepztdep** [reply]
Great! Thank you. It works now. I will upload the computational results once I have completed it.

**Reply by ztdepztdep** [reply]
no, I have tried many parameters combinations, it cann't give any meaningful results. full of "inf".
